homeassistant/components/media_player/services.yaml

```diff
@@ -281,3 +281,13 @@ kodi_call_method:
     method:
       description: Name of the Kodi JSONRPC API method to be called.
       example: 'VideoLibrary.GetRecentlyAddedEpisodes'
+
+squeezebox_call_method:
+  description: 'Call a Squeezebox JSON/RPC API method.'
+  fields:
+    entity_id:
+      description: Name(s) of the Squeexebox entities where to run the API method.
+      example: 'media_player.squeezebox_radio'
+    method:
+      description: Squeezebox JSON/RPC method call, in the form of an array of positional parameters. See 'Command Line Interface' official help page from Logitech for details.
```

Comment:
Separate the method arguments out into their own optional field on the schema.

Reply:
I agree that would be nicer (easier to use), but the API supports hundreds of commands, requiring a different number of parameters.
The parameter p2 of one query can have a completely different meaning as the paramater p2 of another one.
So we can't really name the parameters. We could create param_0, param_1 but... up to how many? Most commands have up to 4 but the count can go up quickly, especially for search commands which accept various criteria.

**Extract of the documentation:**
```General command and query format
The format of the commands, queries and server replies is as follows:
[<playerid>] <p0> <p1> ... <pN> <LF>
where:
<playerid> is the unique identifier for the player, usually (but not guaranteed to be)
the MAC address of the player. Some commands are global to the Squeezebox Server
and do not require a <playerid>. For commands requiring it, a random player will be
selected by the Squeezebox Server if the <playerid> is omitted, and returned in the
server reply. <playerid> may be obtained by using the "player id" or "players" queries.

<p0> through <pN> are positional parameters. Pass a "?" to obtain a value for that
parameter in the Squeezebox Server response (i.e. send a query). Details of the
parameters vary for each command as described below.

Each parameter needs to be encoded using percent-style escaping, the same method
as is used in URLs; for example, "The Clash?" would be encoded as "The%20Clash%3F".
This also applies to <playerid>. In the examples below, the escaping is not show for
readability (except %20 for space).
```

**Some examples:**

`info total artists ?`
Returns the number of artists in the DB. Has to be passed as ['info', 'total', 'artists', '?']

`playlist loadalbum <genre> <artist> <album>`
Puts songs matching the specified genre artist and album criteria on the playlist. Songs previously in the playlist are discarded.
For example: ['playlist', 'loadalbum', 'Rock', 'Abba', '*']

`genres <start> <itemsPerResponse> <taggedParameters>`
This is an example of command accepting tagged parameters. An example would be
['genres', '0', '10', 'search:rock', 'artist_id:5', 'album_id:6', 'year:2003']
This would restrict the search to the genres proposed by artist with ID 5, available on the album with ID 6 and with tracks in year 2003.

As you can see in just 3 examples, parameter p1 can be static strings 'total', 'loadalbum', or a number (index of first result).
I think it is easier for somebody used to the Squeezebox API to just give the array of parameters to the service as is.
Let me know if you have another idea!

If you don't have a squeezebox server running, you can find a copy of the documentation here:
https://github.com/elParaguayo/LMS-CLI-Documentation/blob/master/LMS-CLI.md

Reply:
Ah, sorry, that was a confusing comment. I just meant separate this out into two fields. `method`, which can just be a required string, and `arguments`, which is an optional array of any arguments.

Reply:
OK I get it now.
I split the unique `method` field into two:
* `command` which is a mandatory string and will be the first parameter passed to the API. I use the word `command` to be consistent with the API documentation. I did not use the word 'method' because it would be conflicting with the main API method which is always 'slim.request'
* `parameters` which is an optional array of extra parameters added to the command.